### PR TITLE
Fetch related content for taxonomy sidebar

### PR DESCRIPTION
**Pull request by alecgibson**
We will be querying Rummager for links related to the content we are looking at, scoped to the taxon(s) our current content is tagged to.

This change updates the already-existing `TaxonomySidebar` class to include `related_content` in its hash. There will be a corresponding Pull Request in `static`, where this extra field will be used in the sidebar component.

### Trello

https://trello.com/c/EJXC60k1/429-add-more-like-this-results-to-side-navigation-in-frontend-apps

### CODE REVIEW

File: lib/guidance.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+module Guidance
```
Comment (carvil) [inline lib/guidance.rb]: Let's add a comment discouraging other people from copying this bit of code.

File: lib/govuk_navigation_helpers/taxonomy_sidebar.rb [hunk 2]
```diff
@@ -19,8 +22,20 @@ def taxons
           title: parent_taxon.title,
           url: parent_taxon.base_path,
           description: parent_taxon.description,
+          related_content: content_related_to(parent_taxon),
         }
       end
     end
+
+    def content_related_to(taxon)
```
Comment (carvil) [inline lib/govuk_navigation_helpers/taxonomy_sidebar.rb]: This call cannot fail, could you please make sure we rescue potential issues here?

File: lib/guidance.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  DOCUMENT_TYPES = %w[
```
Comment (carvil) [inline lib/guidance.rb]: Can we also make sure `collections` uses this list from navigation helpers? That way we can delete https://github.com/alphagov/collections/blob/29a0085996311fa7e650dfd65eb805007c2cead9/app/models/rummager_search.rb#L4 and keep just one list of document types until we migrate this to the content store and search

File: lib/govuk_navigation_helpers/taxonomy_sidebar.rb [hunk 4]
```diff
@@ -19,8 +22,20 @@ def taxons
           title: parent_taxon.title,
           url: parent_taxon.base_path,
           description: parent_taxon.description,
+          related_content: content_related_to(parent_taxon),
```
Comment (carvil) [inline lib/govuk_navigation_helpers/taxonomy_sidebar.rb]: Let's also add comments here so people are aware this is a temporary fix that is meant to be migrated to the content store?

File: spec/taxonomy_sidebar_spec.rb [hunk 5]
```diff
@@ -1,7 +1,22 @@
 require 'spec_helper'
+require 'webmock/rspec'
 
 RSpec.describe GovukNavigationHelpers::TaxonomySidebar do
```
Comment (carvil) [inline spec/taxonomy_sidebar_spec.rb]: Let's also test for when we get a bad response from search.

File: lib/guidance.rb (hunk 3 above)
Comment (alecgibson) [inline lib/guidance.rb]: Will do.

**Comment by alecgibson** [comment]
@carvil I've made the requested changes - can you please re-review?

**Comment by carvil** [comment]
@alecgibson this looks fine. Could you please make sure collections now uses this master list rather than keeping a new list there as well?

**Comment by alecgibson** [comment]
@carvil I'll do that now (there's also a task on [the trello card](https://trello.com/c/EJXC60k1/429-add-more-like-this-results-to-side-navigation-in-frontend-apps) to track this)

**Pull request merged**